crates/autopilot/src/infra/persistence/auction/dto.rs

```diff
@@ -0,0 +1,309 @@
+        }
+    }
+}
+
+impl From<boundary::SellTokenSource> for domain::auction::order::SellTokenSource {
+    fn from(source: boundary::SellTokenSource) -> Self {
+        match source {
+            boundary::SellTokenSource::Erc20 => Self::Erc20,
+            boundary::SellTokenSource::External => Self::External,
+            boundary::SellTokenSource::Internal => Self::Internal,
+        }
+    }
+}
+
+impl From<domain::auction::order::BuyTokenDestination> for boundary::BuyTokenDestination {
+    fn from(destination: domain::auction::order::BuyTokenDestination) -> Self {
+        match destination {
+            domain::auction::order::BuyTokenDestination::Erc20 => Self::Erc20,
+            domain::auction::order::BuyTokenDestination::Internal => Self::Internal,
+        }
+    }
+}
+
+impl From<boundary::BuyTokenDestination> for domain::auction::order::BuyTokenDestination {
+    fn from(destination: boundary::BuyTokenDestination) -> Self {
+        match destination {
+            boundary::BuyTokenDestination::Erc20 => Self::Erc20,
+            boundary::BuyTokenDestination::Internal => Self::Internal,
+        }
+    }
+}
+
+impl From<domain::auction::order::AppDataHash> for boundary::AppDataHash {
+    fn from(hash: domain::auction::order::AppDataHash) -> Self {
+        Self(hash.0)
+    }
+}
+
+impl From<boundary::AppDataHash> for domain::auction::order::AppDataHash {
+    fn from(hash: boundary::AppDataHash) -> Self {
+        Self(hash.0)
+    }
+}
+
+impl From<domain::auction::order::Signature> for boundary::Signature {
+    fn from(signature: domain::auction::order::Signature) -> Self {
+        match signature {
+            domain::auction::order::Signature::Eip712(s) => Self::Eip712(s.into()),
+            domain::auction::order::Signature::EthSign(s) => Self::EthSign(s.into()),
+            domain::auction::order::Signature::Eip1271(b) => Self::Eip1271(b),
+            domain::auction::order::Signature::PreSign => Self::PreSign,
+        }
+    }
+}
+
+impl From<boundary::Signature> for domain::auction::order::Signature {
+    fn from(signature: boundary::Signature) -> Self {
+        match signature {
+            boundary::Signature::Eip712(s) => Self::Eip712(s.into()),
+            boundary::Signature::EthSign(s) => Self::EthSign(s.into()),
+            boundary::Signature::Eip1271(b) => Self::Eip1271(b),
+            boundary::Signature::PreSign => Self::PreSign,
+        }
+    }
+}
+
+impl From<domain::auction::order::EcdsaSignature> for boundary::EcdsaSignature {
+    fn from(signature: domain::auction::order::EcdsaSignature) -> Self {
+        Self {
+            r: signature.r,
+            s: signature.s,
+            v: signature.v,
+        }
+    }
+}
+
+impl From<boundary::EcdsaSignature> for domain::auction::order::EcdsaSignature {
+    fn from(signature: boundary::EcdsaSignature) -> Self {
+        Self {
+            r: signature.r,
+            s: signature.s,
+            v: signature.v,
+        }
+    }
+}
```

Comment:
Very large module. Could maybe be broken up into auction, order and mod.rs?

Reply:
divided